### DISCUSSION TITLE
Does Godmode need to be active for commands? +Display Behaviour Question

### DISCUSSION THREAD

**Discussion by Jay-Ti** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#296] [Q&A]
Great Project, got a 4 wire palm springs 54129  modified it to work offsite over MQTT with an URL instead of an IP address, still working on making the commands over MQTT work though(at least i can already read the temperature).
Am i right in my assumption to change settings( as in start the pump, start the heater, start the blower) God mode(in MQTT commands) or Take control(on the webpage need to be active?)
At least right now thats the only way i can start them from the webpage.
At the same time the webpage doesnt show the Display contents, its just empty, while the webpage reports the Temp correctly the display works to switch on the pump and blower, any suggestions on that? Shouldn't the Webpage show the display content and the display show the changed values we changed over the webpage?

TLDR:
Does godmode/Take control need to be active to control pump heater etc over MQTT/webpage or should it work without?
And shouldnt the display show the values we changed over the Webpage/MQTT?

**Comment by visualapproach** [comment]
Hi! On the 4-wire pumps you can only control from the display OR the webpage/mqtt. Not both at the same time. If in godmode, the display cannot control the pump anymore. And there is no way to directly update the display from the esp.

**Reply by chunkysteveo** [reply]
And from my experience - when in "God mode", anyone that messes with the controls then causes the tub to E02 error out.

**Reply by Jay-Ti** [reply]
hmm, also doesn't retain the commands when godmode is deactivated as far as i can tell. Oh well gotta do something with MQTT then.

Started working on this project for myself on the last version, updated now to the latest one, though for whatever reason the ESP keeps rebooting a few times per minute with reset cause 4.
Display and Webpage work fine in that time though ( Well except the Webpage doesnt show what the display shows, but temps and starting the pump/ blower are fine). Controlling functions over the display also works in the short time the ESP is alive.

Power shouldn't be the Problem since I'm connected to my pc watching the serial port.( also crashes without being connected).
Although not connecting the display seems to do the trick, although then i also can't control anything on the pump either.
Restarting the ESP over the Webpage also causes rst cause 4 but i think that's the intended way.


I'm a bit at a loss as to what I could do different / try at the moment.
Any input or help is greatly appreciated right now.

**Reply by visualapproach** [reply]
Go to clean build (in the platformio side panel). Rebuild all and upload via usb. I think april fw could mess up when OTA

**Reply by Jay-Ti** [reply]
> Go to clean build (in the platformio side panel). Rebuild all and upload via usb. I think april fw could mess up when OTA

could just delete the OTA, since i upload over wire atm, but clean build didnt change anything, still crashes with dsp connected. No change there.

**Reply by visualapproach** [reply]
Cause 4 is hw watch dog timer.

**Reply by visualapproach** [reply]
Check wiring. Upload something else (blink sketch). Working? Erase flash and reupload.

**Reply by Jay-Ti** [reply]
> Check wiring. Upload something else (blink sketch). Working? Erase flash and reupload.

Wiring should be fine since commands do get correctly relayed while the esp is alive. But yeah will shorten and resolder the wires in the comming days if i find time.
Erasing everything and re-uploading yields the same results, the moment the spa starts/the dsp gets connected its only a matter of short time until Cause 4 is thrown. If the display is disconnected and the spa is on, Error codes are being read and the esp doesnt crash.
Tried a new ESP8266 same issue. the moment its in contact with the DSP it keeps crashing.

Seems to crash Right around   ESP.wdtFeed(); in BWC::getJSONstates();

**Reply by Jay-Ti** [reply]
So...
I figured out why it kept crashing.
For some reason being connected to my PC caused it to crash.
It got worse when actually having the serial Monitor open.
Having it disconnected it has been working fine for 4 days and counting. 

Idk Voltage missmatch, too much cpu usage keeping the serial port alive, whatever.
